### ISSUE TITLE
After several attempts - no success. Version conflict.

### ISSUE THREAD

**Issue by tapirencrypt** [EU-EDPS/website-evidence-collector#69]
I have now tried it on an Ubuntu PC and on three Raspberry PIs. No success. Unfortunately, I'm a beginner and can't get any further.

```
pi@MORSE:~ $ npm install --global https://github.com/EU-EDPS/website-evidence-collector/tarball/latest
npm WARN ERESOLVE overriding peer dependency
npm WARN Found: puppeteer@2.1.1
npm WARN node_modules/website-evidence-collector/node_modules/puppeteer
npm WARN   puppeteer@"~2.1.0" from website-evidence-collector@1.0.0
npm WARN   node_modules/website-evidence-collector
npm WARN     website-evidence-collector@"https://github.com/EU-EDPS/website-evidence-collector/tarball/latest" from the root project
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer puppeteer@"5.x || 6.x || 7.x || 8.x || 9.x || 10.x || 11.x || 12.x || 13.x" from @cliqz/adblocker-puppeteer@1.23.5
npm WARN node_modules/website-evidence-collector/node_modules/@cliqz/adblocker-puppeteer
npm WARN   @cliqz/adblocker-puppeteer@"^1.9.2" from website-evidence-collector@1.0.0
npm WARN   node_modules/website-evidence-collector
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated request-promise-native@1.0.9: request-promise-native has been deprecated because it extends the now deprecated request package, see https://github.com/request/request/issues/3142
npm WARN deprecated debug@4.1.1: Debug versions >=3.2.0 <3.2.7 || >=4 <4.3.1 have a low-severity ReDos regression when used in a Node.js environment. It is recommended you upgrade to 3.2.7 or 4.3.1. (https://github.com/visionmedia/debug/issues/797)
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated puppeteer@2.1.1: Version no longer supported. Upgrade to @latest
npm WARN deprecated core-js@2.6.12: core-js@<3.4 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Please, upgrade your dependencies to the actual version of core-js.

added 256 packages, and audited 257 packages in 1m

13 packages are looking for funding
  run `npm fund` for details

2 high severity vulnerabilities

Some issues need review, and may require choosing
a different dependency.

Run `npm audit` for details.
pi@MORSE:~ $ uname -a
Linux MORSE 5.10.92-v8+ #1514 SMP PREEMPT Mon Jan 17 17:39:38 GMT 2022 aarch64 GNU/Linux
pi@MORSE:~ $ lsb_release -d
Description:	Debian GNU/Linux 11 (bullseye)
pi@MORSE:~ $ npm -v
7.5.2
pi@MORSE:~ $ node -v
v12.22.5
pi@MORSE:~ $ 
```

**Comment by rriemann** [comment]
Hi @tapirencrypt,

thank you for your interest.

I could not reproduce this installation error with:

~~~bash
sudo apt install npm
lsb_release -d
# Description:    Ubuntu 20.04.3 LTS

mkdir "${HOME}/.npm-packages"
npm config set prefix "${HOME}/.npm-packages"
npm install --global https://github.com/EU-EDPS/website-evidence-collector/tarball/latest
~~~

However, we have been working in the previous months on the preparation of a new release with updated dependencies. We just did not release a new version because we thought to receive first some user feedback on the testing version. So for we have not heard of issues.

Please let me know if you can reproduce the problem with the latest testing version (main/master):

> npm install --global https://github.com/EU-EDPS/website-evidence-collector/tarball/master

**Comment by tapirencrypt** [comment]
Hi,

Unfortunately, it does not work on the Raspbbery PI.
I'll try it on an Ubuntu PC

bye

***@***.***:~ $ sudo apt install npm
Paketlisten werden gelesen… Fertig
Abhängigkeitsbaum wird aufgebaut… Fertig
Statusinformationen werden eingelesen… Fertig
npm ist schon die neueste Version (7.5.2+ds-2).
0 aktualisiert, 0 neu installiert, 0 zu entfernen und 6 nicht aktualisiert.
***@***.***:~ $ lsb_release -d
Description: Debian GNU/Linux 11 (bullseye)
***@***.***:~ $ mkdir "${HOME}/.npm-packages"
mkdir: das Verzeichnis „/home/pi/.npm-packages“ kann nicht angelegt werden: Die Datei existiert bereits
***@***.***:~ $ npm config set prefix "${HOME}/.npm-packages"
***@***.***:~ $ npm install --global https://github.com/EU-EDPS/website-evidence-collector/tarball/latest
npm WARN ERESOLVE overriding peer dependency
npm WARN Found: ***@***.***
npm WARN node_modules/website-evidence-collector/node_modules/puppeteer
npm WARN puppeteer@"~2.1.0" from ***@***.***
npm WARN node_modules/website-evidence-collector
npm WARN website-evidence-collector@"https://github.com/EU-EDPS/website-evidence-collector/tarball/latest" from the root project
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer puppeteer@"5.x || 6.x || 7.x || 8.x || 9.x || 10.x || 11.x || 12.x || 13.x" from @***@***.***
npm WARN ***@***.***/adblocker-puppeteer
npm WARN @cliqz/adblocker-puppeteer@"^1.9.2" from ***@***.***
npm WARN node_modules/website-evidence-collector
npm WARN deprecated ***@***.***: request-promise-native has been deprecated because it extends the now deprecated request package, see https://github.com/request/request/issues/3142
npm WARN deprecated ***@***.***: this library is no longer supported
npm WARN deprecated ***@***.***: Debug versions >=3.2.0 <3.2.7 || >=4 <4.3.1 have a low-severity ReDos regression when used in a Node.js environment. It is recommended you upgrade to 3.2.7 or 4.3.1. (https://github.com/visionmedia/debug/issues/797)
npm WARN deprecated ***@***.***: Please upgrade to version 7 or higher. Older versions may use Math.random() in certain circumstances, which is known to be problematic. See https://v8.dev/blog/math-random for details.
npm WARN deprecated ***@***.***: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated ***@***.***: Version no longer supported. Upgrade to @latest
npm WARN deprecated ***@***.***: core-js@<3.4 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Please, upgrade your dependencies to the actual version of core-js.

changed 256 packages, and audited 257 packages in 1m

13 packages are looking for funding
run `npm fund` for details

2 high severity vulnerabilities

Some issues need review, and may require choosing
a different dependency.

Run `npm audit` for details.
***@***.***:~ $

------- Original Message -------
Robert Riemann ***@***.***> schrieb am Montag, 14. Februar 2022 um 17:48:

**Comment by tapirencrypt** [comment]
The same on Ubuntu.

fmzt:~$ sudo apt install npm
[sudo] Passwort für markus:
Paketlisten werden gelesen… Fertig
Abhängigkeitsbaum wird aufgebaut… Fertig
Statusinformationen werden eingelesen… Fertig
npm ist schon die neueste Version (7.5.2+ds-2).
0 aktualisiert, 0 neu installiert, 0 zu entfernen und 0 nicht aktualisiert.
fmzt:~$ lsb_release -d
Description: Ubuntu 21.10
fmzt:~$ mkdir "${HOME}/.npm-packages"
fmzt:~$ npm config set prefix "${HOME}/.npm-packages"
fmzt:~$ npm install --global https://github.com/EU-EDPS/website-evidence-collector/tarball/latest
npm WARN ERESOLVE overriding peer dependency
npm WARN Found: ***@***.***
npm WARN node_modules/website-evidence-collector/node_modules/puppeteer
npm WARN puppeteer@"~2.1.0" from ***@***.***
npm WARN node_modules/website-evidence-collector
npm WARN website-evidence-collector@"https://github.com/EU-EDPS/website-evidence-collector/tarball/latest" from the root project
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer puppeteer@"5.x || 6.x || 7.x || 8.x || 9.x || 10.x || 11.x || 12.x || 13.x" from @***@***.***
npm WARN ***@***.***/adblocker-puppeteer
npm WARN @cliqz/adblocker-puppeteer@"^1.9.2" from ***@***.***
npm WARN node_modules/website-evidence-collector
npm WARN deprecated ***@***.***: this library is no longer supported
npm WARN deprecated ***@***.***: Debug versions >=3.2.0 <3.2.7 || >=4 <4.3.1 have a low-severity ReDos regression when used in a Node.js environment. It is recommended you upgrade to 3.2.7 or 4.3.1. (https://github.com/visionmedia/debug/issues/797)
npm WARN deprecated ***@***.***: request-promise-native has been deprecated because it extends the now deprecated request package, see https://github.com/request/request/issues/3142
npm WARN deprecated ***@***.***: Please upgrade to version 7 or higher. Older versions may use Math.random() in certain circumstances, which is known to be problematic. See https://v8.dev/blog/math-random for details.
npm WARN deprecated ***@***.***: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated ***@***.***: Version no longer supported. Upgrade to @latest
npm WARN deprecated ***@***.***: core-js@<3.4 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Please, upgrade your dependencies to the actual version of core-js.

added 256 packages, and audited 257 packages in 37s

13 packages are looking for funding
run `npm fund` for details

2 high severity vulnerabilities

Some issues need review, and may require choosing
a different dependency.

Run `npm audit` for details.
fmzt:~$

------- Original Message -------
Robert Riemann ***@***.***> schrieb am Montag, 14. Februar 2022 um 17:48:

**Comment by rriemann** [comment]
Hi!

I see that you in both cases used the latest release. Can you please try again with `master`?

~~~bash
npm install --global https://github.com/EU-EDPS/website-evidence-collector/tarball/master
~~~

**Comment by tapirencrypt** [comment]
Sorry it not work

***@***.***:~ $ npm install --global https://github.com/EU-EDPS/website-evidence-collector/tarball/master
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE package: ***@***.***',
npm WARN EBADENGINE required: { node: '>=14.0' },
npm WARN EBADENGINE current: { node: 'v12.22.5', npm: '7.5.2' }
npm WARN EBADENGINE }
npm WARN deprecated ***@***.***: Debug versions >=3.2.0 <3.2.7 || >=4 <4.3.1 have a low-severity ReDos regression when used in a Node.js environment. It is recommended you upgrade to 3.2.7 or 4.3.1. (https://github.com/visionmedia/debug/issues/797)
npm WARN deprecated ***@***.***: Version no longer supported. Upgrade to @latest

added 49 packages, removed 97 packages, changed 159 packages, and audited 209 packages in 46s

25 packages are looking for funding
run `npm fund` for details

3 high severity vulnerabilities

Some issues need review, and may require choosing
a different dependency.

Run `npm audit` for details.
***@***.***:~ $

------- Original Message -------
Robert Riemann ***@***.***> schrieb am Donnerstag, 17. Februar 2022 um 22:22:

**Comment by rriemann** [comment]
Dear @tapirencrypt ,

I think the error log here is quite clear:

> npm WARN EBADENGINE Unsupported engine {
> npm WARN EBADENGINE package: ***@***.***',
> npm WARN EBADENGINE required: { node: '>=14.0' },
> npm WARN EBADENGINE current: { node: 'v12.22.5', npm: '7.5.2' }

You have node in version 12.22.5. However, WEC requires node in version 14.0 or higher. The installation section in the [Readme](https://github.com/EU-EDPS/website-evidence-collector#installation) say so in step 1:

>  The minimum version for Node.js is 14.0.

Please update your node version and try again.

**Comment by tapirencrypt** [comment]
***@***.***:~# apt-get install -y nodejs
Paketlisten werden gelesen... Fertig
Abhängigkeitsbaum wird aufgebaut.
Statusinformationen werden eingelesen.... Fertig
Die folgenden Pakete werden aktualisiert (Upgrade):
nodejs
1 aktualisiert, 0 neu installiert, 0 zu entfernen und 0 nicht aktualisiert.
Es müssen 27,0 MB an Archiven heruntergeladen werden.
Nach dieser Operation werden 85,6 MB Plattenplatz zusätzlich benutzt.
Holen:1 https://deb.nodesource.com/node_17.x buster/main amd64 nodejs amd64 17.6.0-deb-1nodesource1 [27,0 MB]
Es wurden 27,0 MB in 2 s geholt (10,9 MB/s).
apt-listchanges: Changelogs werden gelesen...
(Lese Datenbank ... 232671 Dateien und Verzeichnisse sind derzeit installiert.)
Vorbereitung zum Entpacken von .../nodejs_17.6.0-deb-1nodesource1_amd64.deb ...
Entpacken von nodejs (17.6.0-deb-1nodesource1) über (12.22.10-deb-1nodesource1) ...
nodejs (17.6.0-deb-1nodesource1) wird eingerichtet ...
Trigger für man-db (2.8.5-2) werden verarbeitet ...
***@***.***:~# node -v
v17.6.0
***@***.***:~# npm -v
8.5.1
***@***.***:~# npm install --global https://github.com/EU-EDPS/website-evidence-collector/tarball/master
npm WARN deprecated ***@***.***: Debug versions >=3.2.0 <3.2.7 || >=4 <4.3.1 have a low-severity ReDos regression when used in a Node.js environment. It is recommended you upgrade to 3.2.7 or 4.3.1. (https://github.com/visionmedia/debug/issues/797)
npm WARN deprecated ***@***.***: Version no longer supported. Upgrade to @latest

added 208 packages, and audited 209 packages in 41s

25 packages are looking for funding
run `npm fund` for details

2 high severity vulnerabilities

To address all issues (including breaking changes), run:
npm audit fix --force

Run `npm audit` for details.
npm notice
npm notice New patch version of npm available! 8.5.1 -> 8.5.2
npm notice Changelog: https://github.com/npm/cli/releases/tag/v8.5.2
npm notice Run npm install -g ***@***.*** to update!
npm notice
***@***.***:~# website-evidence-
website-evidence-collector website-evidence-reporter
***@***.***:~# website-evidence-collector https://meldekopf.de/
/usr/lib/node_modules/website-evidence-collector/node_modules/puppeteer/lib/cjs/puppeteer/node/BrowserRunner.js:197
reject(new Error([
^

Error: Failed to launch the browser process!
[0225/113749.187211:ERROR:zygote_host_impl_linux.cc(90)] Running as root without --no-sandbox is not supported. See https://crbug.com/638180.

TROUBLESHOOTING: https://github.com/puppeteer/puppeteer/blob/main/docs/troubleshooting.md

at onClose (/usr/lib/node_modules/website-evidence-collector/node_modules/puppeteer/lib/cjs/puppeteer/node/BrowserRunner.js:197:20)
at Interface. (/usr/lib/node_modules/website-evidence-collector/node_modules/puppeteer/lib/cjs/puppeteer/node/BrowserRunner.js:187:68)
at Interface.emit (node:events:539:35)
at Interface.close (node:internal/readline/interface:529:10)
at Socket.onend (node:internal/readline/interface:258:10)
at Socket.emit (node:events:539:35)
at endReadableNT (node:internal/streams/readable:1342:12)
at processTicksAndRejections (node:internal/process/task_queues:83:21)

Node.js v17.6.0
***@***.***:~# ls
output saltscript.sh test.html test.yaml y2h.py y-h.py
***@***.***:~#

------- Original Message -------
Robert Riemann ***@***.***> schrieb am Montag, 21. Februar 2022 um 12:55:
